Lab kinematic results - local processing

antoinefalisse opened the pull request:
This PR is meant to enable users to regenerate the paper results after downloading data from SimTK. The previous code was pulling data from the database, but the sessions aren't public so this was only possible for admins. This is also a good example of how to process data locally.

The dataset on SimTK should be slightly adjusted:
- The videos for computing the extrinsics are not there
- A couple other minor things

Everything is ready [here](https://drive.google.com/drive/folders/1VuhbZD3ZRFcGYkDU4IOFl8HC9t2kIleU?usp=sharing)
I adjusted the metadata to remove personal info. Maybe you want to take a quick look.
I will upload the datasets to SimTK when the PR is merged.

Also, the dataset on SimTK is structured differently as compared to what main expects, as part of labValidationVideosToKinematics, there is now a piece of code to re-structure the data. It basically copies over what is needed from the dataset to a different location and the processing is done from there. That means the newly processed data will not overwrite or be in the same folder as the dataset from SimTK, which is fine I guess.

I ran the code locally for one subject, and it worked fine. Let's ask Matt to test it (it can be done after merging).
Since I changed a couple of things in main and utilsDetector, I will do a proper testing of the app when merged in dev.

suhlrich commented on ReproducePaperResults/labValidationVideosToKinematics.py:
```diff
@@ -40,143 +43,225 @@
+# To reprocess the data, we need to re-organize the data so that the folder
+# structure is the same one as the one expected by OpenCap. It is only done
+# once as long as the variable overwriteRestructuring is False. To overwrite
+# filpt the flag to True.
```
flip the flag?

suhlrich commented on ReproducePaperResults/labValidationVideosToKinematics.py:
```diff
@@ -40,143 +43,225 @@
+                
+                # Session specific intrinsic parameters
+                if 'subject2' in sessionName or 'subject3' in sessionName:
+                    intrinsicsFinalFolder = 'Deployed_720_240fps'
```
these videos are 60fps. Am I forgetting something?

antoinefalisse replied:
The first two subjects were at 240fps

antoinefalisse replied:
First three actually
https://github.com/stanfordnmbl/opencap-core/blob/main/ReproducePaperResults/labValidationIDs.py#L14

suhlrich commented on utilsDetector.py:
```diff
@@ -37,20 +37,25 @@ def runPoseDetector(CameraDirectories, trialRelativePath, pathPoseDetector,
                 resolutionPoseDetection=resolutionPoseDetection,
                 generateVideo=generateVideo)
         elif poseDetector == 'mmpose':
             runMMposeVideo(
```
seems we don't get the extension if we are using mmpose?

antoinefalisse replied:
Fixed in latest commit

antoinefalisse commented:
Thanks for the review - I addressed the comments.
I will make a proper testing on Ubuntu with HRNet later if you don't mind.
Will test dev pipeline with both openpose and hrnet when merged.

suhlrich commented on ReproducePaperResults/labValidationVideosToKinematics.py:
```diff
@@ -264,4 +264,6 @@ def process_trial(trial_name=None, session_name=None, isDocker=False,
\ No newline at end of file
+                              dataDir=dataDir)
+                
+            test=1
```
remove

suhlrich reviewed (commented):
looks good! ready for matt testing
